### PR TITLE
reduce allocations in dims_howmany

### PR DESCRIPTION
**Pull request by jishnub**
This reduces some unnecessary allocations in `dims_howmany`. This may be reduced further  by using `StaticArrays`, but that's a separate discussion as it'll increase the load time.

On master
```julia
julia> X = rand(100, 100);

julia> @btime FFTW.dims_howmany($X, $X, $(collect(size(X))), 1)
  689.000 ns (21 allocations: 1.59 KiB)
([100; 1; 1;;], [100; 100; 100;;])
```
This PR
Updated after https://github.com/JuliaMath/FFTW.jl/pull/269/commits/e15a9b8c8576a714f95d70cf32edf093c0b647a1
```julia
julia> @btime FFTW.dims_howmany($X, $X, size($X), 1);
  73.707 ns (2 allocations: 160 bytes)
```

### CODE REVIEW

File: src/fft.jl [hunk 1]
```diff
@@ -566,19 +566,19 @@ unsafe_execute!(plan::r2rFFTWPlan{T},
...
-    ost = [strides(Y)...]
-    dims = Matrix(transpose([sz[reg] ist[reg] ost[reg]]))
-    oreg = [1:ndims(X);]
-    oreg[reg] .= 0
-    oreg = filter(d -> d > 0, oreg)
-    howmany = Matrix(transpose([sz[oreg] ist[oreg] ost[oreg]]))
+    ist = _collect_Intvector(strides(X))
+    ost = _collect_Intvector(strides(Y))
+    dims = vcat(view(sz, reg)', view(ist, reg)', view(ost, reg)')
+    oreg = [i for i in 1:ndims(X) if i ∉ reg]
```
Comment (stevengj) [inline src/fft.jl:580]:
```suggestion
    oreg = filter(∉(reg), 1:ndims(X))
```

**Comment by jishnub** [comment]
Thanks! I've updated it now, and all the unnecessary allocations are now avoided if `region` is an `Int` or a `Tuple`

**Comment by jishnub** [comment]
This should be ready now

**Comment by jishnub** [comment]
Gentle bump

**Comment by jishnub** [comment]
#266 should ideally have been v1.7.0 instead of v1.6.1 as it is mildly breaking, so I'm bumping the version in this PR instead.

**Comment by jishnub** [comment]
Gentle bump @stevengj

**Comment by stevengj** [comment]
LGTM

**Pull request merged**